src/main/java/pl/edu/agh/firecell/renderer/BasicRenderer.java

```diff
@@ -32,9 +43,82 @@ public BasicRenderer(float aspectRatio, IOListener ioListener, SimulationConfig
     @Override
     public void render(State state, double frameTime) {
         cameraController.update(frameTime);
-        shader.bind();
-        shader.setMatrix4("uView", camera.viewMatrix());
-        renderState(state);
+        opaqueMaterialShader.bind();
+        opaqueMaterialShader.setMatrix4("uView", camera.viewMatrix());
+        transparentTempShader.bind();
+        transparentTempShader.setMatrix4("uView", camera.viewMatrix());
+        fireShader.bind();
+        fireShader.setMatrix4("uView", camera.viewMatrix());
+
+        switch (renderMode) {
```

Comment:
Maybe use strategy pattern with this rendering mode to clean up code in renderer